projects/plugins/jetpack/modules/search/instant-search/lib/date.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Get user friendly date string from y-m-d h:i:s to Apr 3, 2020
+ *
+ * @param {string} dateTime - y-m-d h:i:s date string
+ * @returns {string} - human friendly date string
+ */
+export const getDateString = dateTime => {
+	const dateTimeParts = dateTime.split( ' ' );
+	if ( ! Array.isArray( dateTimeParts ) || dateTimeParts.length !== 2 ) {
+		return dateTime;
+	}
+
+	const dateParts = dateTimeParts[ 0 ].split( '-' );
+	if ( ! Array.isArray( dateParts ) || dateParts.length !== 3 ) {
+		return dateTime;
+	}
+
+	const year = dateParts[ 0 ];
+	const month = dateParts[ 1 ];
+	const day = dateParts[ 2 ];
+
+	let monthString = month;
+	switch ( parseInt( month ) ) {
+		case 1:
+			monthString = __( 'Jan', 'jetpack' );
+			break;
+		case 2:
+			monthString = __( 'Feb', 'jetpack' );
+			break;
+		case 3:
+			monthString = __( 'Mar', 'jetpack' );
+			break;
+		case 4:
+			monthString = __( 'Apr', 'jetpack' );
+			break;
+		case 5:
+			monthString = __( 'May', 'jetpack' );
+			break;
+		case 6:
+			monthString = __( 'Jun', 'jetpack' );
+			break;
+		case 7:
+			monthString = __( 'Jul', 'jetpack' );
+			break;
+		case 8:
+			monthString = __( 'Aug', 'jetpack' );
+			break;
+		case 9:
+			monthString = __( 'Sep', 'jetpack' );
+			break;
+		case 10:
+			monthString = __( 'Oct', 'jetpack' );
+			break;
+		case 11:
+			monthString = __( 'Nov', 'jetpack' );
+			break;
+		case 12:
+			monthString = __( 'Dec', 'jetpack' );
+			break;
+	}
+
+	return `${ monthString } ${ day }, ${ year }`;
```

Comment:
I'm not sure this will work well across all languages + locales. Can we use something like `toLocaleDateString` instead? 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleDateString

Reply:
There is also `dateI18n` from `@wordpress/date`, which will take the site's date format into account.

`@wordpress/date` as a whole may allow yo to simplify all this:
https://developer.wordpress.org/block-editor/reference-guides/packages/packages-date/

Reply:
One caveat: Using `@wordpress/date` will bloat our bundle size by [103 kB gzipped (!)](https://bundlephobia.com/package/@wordpress/date@4.1.1) and is likely not a viable solution here.

Reply:
Indeed, I did not want to introduce new dependencies, will look into using the built in support, thanks for the suggestion.

Reply:
Should be fixed now.